### PR TITLE
fix: Rename database from 'couchbasedb' to 'couchbase' in documentation and db_engine_specs

### PR DESCRIPTION
**Pull request by ayush-couchbase**
### SUMMARY
Changing Name from couchbasedb to couchbase. Makes it more clear and logical about underlying database.
No major change done.
### TESTING INSTRUCTIONS
mannual testing

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/sql_parse.py
Comment (villebro) [inline superset/sql_parse.py:105]: This would entail a breaking change, and would break every Couchbase connection created by existing users. At minimum we would need to create an alias for the old dialect name to ensure old connections still work.

**Review by villebro — CHANGES_REQUESTED** [review]
There are some issues reported by CI. In addition, the breaking change is problematic, and would need to at least be addressed in `UPDATING.md` and/or by adding an alias to the old name.

File: superset/sql_parse.py
Comment (ayush-couchbase) [reply superset/sql_parse.py:106]: I have corrected this in new patch. I am adding alias to old name also. I am seeing all checks passed too.

Comment (villebro) [reply superset/sql_parse.py:106]: > I am adding alias to old name also.

I don't see the alias here - can you check how the alias is implemented in the Postgres spec?

**Review by villebro — COMMENTED** [review]
Second pass comment

File: superset/sql_parse.py
Comment (ayush-couchbase) [reply superset/sql_parse.py:105]: My mistake I thought alias means updating sql_parse , I saw oceanbase and postgres db_engine_specs and similar to that I have added `engine_aliases = {"couchbase", "couchbasedb"}` to db_engine_spec/couchbase.py .

File: superset/db_engine_specs/couchbase.py
Comment (villebro) [inline superset/db_engine_specs/couchbase.py:79]: You donät need the main name in the alias set, this should be enough:
```suggestion
    engine_aliases = {"couchbasedb"}
```

**Review by villebro — COMMENTED** [review]
One last nit, after that this LGTM 👍

File: superset/db_engine_specs/couchbase.py
Comment (ayush-couchbase) [reply superset/db_engine_specs/couchbase.py]: Updated

Comment (ayush-couchbase) [reply superset/db_engine_specs/couchbase.py]: There are 2 failed checks, but they were not failing earlier, are they because of my patch ?

Comment (villebro) [reply superset/db_engine_specs/couchbase.py]: The failed MySQL test is flaky - we can restart the test. The docs one seems unrelated - It seems `resources/INTHEWILD.md` has a link to `/www.cnovit.com` that isn't working. I think we may need to remove the link if the homepage isn't up anymore. Do you want to do it in this PR? I can also open a PR to fix it if needed.

**Comment by villebro** [comment]
@michael-s-molina this will not be a breaking change as per the last changes. So I think it's safe to remove that label.

File: superset/db_engine_specs/couchbase.py
Comment (ayush-couchbase) [reply superset/db_engine_specs/couchbase.py]: Fixed

File: RESOURCES/INTHEWILD.md
Comment (rusackas) [inline RESOURCES/INTHEWILD.md:83]: Is this removal intentional? Maybe you meant to add Couchbase?

Comment (ayush-couchbase) [reply RESOURCES/INTHEWILD.md:83]: This link is not working. So tests were failing. @villebro is discussing about it in previous conversation.

Comment (villebro) [reply RESOURCES/INTHEWILD.md:83]: @ayush-couchbase it seems the link was failing due to being `http` and not `https` - could you update that? The `https` one works for me at least. After that I think we're good to go.

Comment (ayush-couchbase) [reply RESOURCES/INTHEWILD.md:83]: Sure , Updated in new patch

**Review by villebro — APPROVED** [review]
LGTM, thanks for all the patience and iterations!

File: RESOURCES/INTHEWILD.md
Comment (rusackas) [reply RESOURCES/INTHEWILD.md:83]: FYI, the link checker does not block merging... it just gives us little to-do items as maintainers. I'll open a PR today to sweep up whatever's left at the moment.

**Comment by rusackas** [comment]
FYI @sadpandajoe the author would like to get this into the next release opportunity. I tagged it as 4.1 in case there's an RC4, but let me know if there's anything else I should do to get it into 4.1.1 if RC3 passes the vote.

**Pull request merged**
